### DISCUSSION TITLE
Ability to handle more than 8 peers

### DISCUSSION THREAD

**Discussion by Henac** [insidegui/MultipeerKit#26] [Q&A] [answered]
Thanks for developing this awesome library, it is so much cleaner/swifty. 

Do you know if it is possible to use this to connect to more than 8 devices at one time? 
I was wondering if there was a way to create two instances of the Tranceiver and then handle 8 devices with one tranceiver and 8 devices with another. 

Seeing as you seem to have much more experience with MultipeerConnectivity, I am wondering if you have any thoughts on whether it is currently possible or whether the library could be modified to handle more peers?

Thanks again
Phil

**Comment by insidegui** [comment, marked as answer]
Hi Phil,

Thank you for the kind words. 

I suspect the 8 peer limit is imposed by a physical limitation of the underlying AWDL transport used by MultipeerConnectivity, so it's likely not something that can be fixed in software. 

Something you could try, but which I've never tried myself, would be to multiplex the transceiver, connecting and disconnecting peers in between data exchanges to allow the overflowing peers to come in. 

This wouldn't work for continuous streams of data, but it might work for discrete message exchanges.

The reality is that wireless is hard and there's no magic in it, so we're still limited to what the physical radio waves are capable of handling 😄

**Reply by Henac** [reply]
Thanks. I will attempt the multiplex idea. I think I saw a comment from one of the developers of p2pkit.io state something to that affect, of reducing the number of concurrent connections. 
I appreciate the advice. I will let you know how I go.
